### PR TITLE
Configure and use new eslint package

### PR DESCRIPTION
**Pull request by JorikSchellekens**

### CODE REVIEW

**Comment by JorikSchellekens** [comment]
ts-lint failure is expected

File: src/client.js [hunk 1]
```diff
@@ -3797,7 +3797,7 @@ MatrixClient.prototype.paginateEventTimeline = function(eventTimeline, opts) {
         return pendingRequest;
     }
 
-    let path, params, promise;
+    let path; let params; let promise;
```
Comment (turt2live) [inline src/client.js:3800]: newlines please

File: src/client.js [hunk 2]
```diff
@@ -4217,7 +4217,7 @@ MatrixClient.prototype.getRoomPushRule = function(scope, roomId) {
  */
 MatrixClient.prototype.setRoomMutePushRule = function(scope, roomId, mute) {
     const self = this;
-    let deferred, hasDontNotifyRule;
+    let deferred; let hasDontNotifyRule;
```
Comment (turt2live) [inline src/client.js:4220]: newlines

File: src/index.ts [hunk 3]
```diff
@@ -14,19 +14,20 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
-import * as matrixcs from "./matrix";
-import * as utils from "./utils";
-import request from "request";
+import * as matrixcs from './matrix';
+import * as utils from './utils';
+import request from 'request';
```
Comment (turt2live) [inline src/index.ts:19]: this feels a bit overly strict. Is it just an IDE thing or a requirement of the rules?

Edit: I'm really hoping it's an IDE thing given a good chunk of the diff is quote changes.

File: src/models/event-timeline-set.js [hunk 4]
```diff
@@ -647,7 +647,7 @@ EventTimelineSet.prototype.compareEventOrdering = function(eventId1, eventId2) {
     if (timeline1 === timeline2) {
         // both events are in the same timeline - figure out their
         // relative indices
-        let idx1, idx2;
+        let idx1; let idx2;
```
Comment (turt2live) [inline src/models/event-timeline-set.js:650]: newlines

File: src/realtime-callbacks.js [hunk 5]
```diff
@@ -185,8 +185,8 @@ function _runCallbacks() {
  */
 function binarySearch(array, func) {
     // min is inclusive, max exclusive.
-    let min = 0,
-        max = array.length;
+    let min = 0;
+        let max = array.length;
```
Comment (turt2live) [inline src/realtime-callbacks.js:189]: wrong indentation

**Comment by JorikSchellekens** [comment]
The quote changes were an accident

File: src/index.ts (hunk 3 above)
Comment (JorikSchellekens) [reply src/index.ts:19]: It's an IDE thing. The eslint explicitly turns off quote checks.

**Pull request merged**